posts/2021-07-08-sig-bugs.adoc

```diff
@@ -0,0 +1,133 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17593[EJB Singleton Lifecycle Deadlock]
++
+When the Singleton called another managed object (like another session bean or asynchronous bean) OR the Singleton had a lifecycle interceptor.
+The other managed object calls the Singleton. (Either through an asynchronous method that is invoked at the right time (so a timing problem) or through the method the Singleton called on it).
+This would cause a deadlock because the Singleton would be locked from the lifecycle event.
+
+For `@PreDestroy` we are no longer locking the singleton, depending on when the destroy resolves, the other managed object might get an error stating the bean has been stopped/destroyed.
+
+For `@PostConstruct` the `EJBContainer` must lock the creation of `SingletonBeans` to ensure only one gets created. Instead of deadlocking forever we now have a timeout value of attempting to obtain a lock. This value is 2 minutes by default but configurable by `@AccessTimeout` value either on the class or the `@PostConstruct` method. A thread that times out trying to access the Singleton will get a `javax.ejb.ConcurrentAccessTimeoutException`
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17235[Improve featureUtility return code when invalid action name is specified]
++
+Users would see an erroneous `return code 0` when an invalid action was specified. This was now changed to `BAD_ARGUMENT (20)`. Because of this change any scripted invocations of featureUtility can now detect this.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17299[Allow multiple version of singleton feature with featureUtility installFeature command]
++
+FeatureUtility installFeature command did not support the installation of multiple versions of the singleton feature using a single command. As a workaround, users had to install each version of the feature one by one. The fix will now allow users to install multiple versions of the feature in a single command, and will also allow users to install a feature bundle addon (libertyCoreBundle, baseBundle, etc) to install all features that apply to their liberty edition. 
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17666[JavaMail tries to use a resource file that only exists in the implementation]
++
+Instead of the `javax.mail.Session` constructor (from `wlp/dev/api/spec/com.ibm.websphere.javaee.mail.1.6_1.0.53.jar`) tried to load the resource `/META-INF/javamail.default.address.map`, that is required. However, this file only existed in the implementation (`wlp/lib/com.ibm.ws.com.sun.mail.javax.mail.1.6_1.6.53.jar`) and it is not able to find it, so it caused a warning to show up:
```

Comment:
This sentence doesn't make sense to me.  Maybe something like this instead:
The `javax.mail.Session` constructor (from `wlp/dev/api/spec/com.ibm.websphere.javaee.mail.1.6_1.0.53.jar`) tried to load the resource `/META-INF/javamail.default.address.map`. However, this file only existed in the implementation (`wlp/lib/com.ibm.ws.com.sun.mail.javax.mail.1.6_1.6.53.jar`). It is not able to find it, so it caused a warning to show up:

Reply:
Done